mifospay/src/main/java/org/mifos/mobilewallet/mifospay/history/ui/HistoryFragment.java

```diff
@@ -177,5 +181,37 @@ public void showHistoryFetchingProgress() {
         vStateView.setVisibility(View.GONE);
         rvHistory.setVisibility(View.GONE);
         pbHistory.setVisibility(View.VISIBLE);
+        scrollDownButton.setVisibility(View.GONE);
+    }
+
+    @Override
+    public void setUpScrollDownBottomButton() {
+        rvHistory.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                final int length = mHistoryAdapter.getItemCount();
+                super.onScrollStateChanged(recyclerView, newState);
+                if (!recyclerView.canScrollVertically(1)) {
+                    Toaster.showToast(getContext(), getString(R.string.end_of_transactions));
+                    scrollDownButton.setImageResource(R.drawable.ic_up_arrow);
+                    scrollDownButton.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View view) {
+                            rvHistory.smoothScrollToPosition(0);
+                        }
+                    });
+                } else if (!recyclerView.canScrollVertically(recyclerView.getLayoutParams().height))
+                {   scrollDownButton.setImageResource(R.drawable.ic_down_arrow);
```

Comment:
In here you have a check style violation, the '{' at column 204 should be on the previous line.